frontend/src/lib/SegmentMetricsManager.ts

```diff
@@ -111,10 +116,55 @@ export class SegmentMetricsManager implements MetricsManager {
     this.send(evName, evData)
   }
 
+  public handleDeltaMessage(delta: Delta, metadata: ForwardMsgMetadata): void {
```

Comment:
nit: I believe the delta stats event (which requires the delta counter) is actually unused and planned to be removed by the data team. So, maybe we can already remove it right now. Probably something to sync with @blackary.

Reply:
Nice. I'm going to merge this as-is (because it unblocks the StreamlitLib project), and then work on a followup after chatting with @blackary.